src/Sylius/Bundle/ShopBundle/EventListener/AlreadyLoggedInUserRegistrationListener.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ShopBundle\EventListener;
+
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+final class AlreadyLoggedInUserRegistrationListener implements EventSubscriberInterface
+{
+    private AuthorizationCheckerInterface $authorizationChecker;
+
+    private RequestStack $requestStack;
+
+    private RouterInterface $router;
+
+    public function __construct(
+        AuthorizationCheckerInterface $authorizationChecker,
+        RequestStack $requestStack,
+        RouterInterface $router
+    ) {
+        $this->authorizationChecker = $authorizationChecker;
+        $this->requestStack = $requestStack;
+        $this->router = $router;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'sylius.customer.initialize_register' => 'handleAlreadyConnectedUser',
+        ];
+    }
+
+    public function handleAlreadyConnectedUser(ResourceControllerEvent $event): void
+    {
+        if (!$this->authorizationChecker->isGranted('IS_AUTHENTICATED_FULLY')) {
```

Comment:
```suggestion
        if (!$this->authorizationChecker->isGranted('IS_AUTHENTICATED_REMEMBERED')) {
```

Please check for less strict role
![image](https://github.com/Sylius/Sylius/assets/870747/fc3c3631-44b4-4e0e-b210-eb8426755209)

https://symfony.com/doc/5.4/security.html#checking-to-see-if-a-user-is-logged-in

Reply:
There is a caveat with 6.3, they have defined new "role" `IS_AUTHENTICATED`, but old will still work and I don't know of the way to make code future proof right now.
 
![image](https://github.com/Sylius/Sylius/assets/870747/1c4f38db-fd36-4573-89a5-361c4593c208)

https://symfony.com/doc/current/security.html#checking-to-see-if-a-user-is-logged-in

Reply:
Please update role `IS_AUTHENTICATED_REMEMBERED` for login as well.

https://github.com/Sylius/Sylius/blob/39cb67c9980fa1e3b676a917be07e32d109a16e1/src/Sylius/Bundle/UiBundle/Controller/SecurityController.php#L41-L43

Reply:
Yep, I wanted to stick with what there is in the SecurityController, but I guess we can change for `IS_AUTHENTICATED_REMEMBERED` :+1: thanks !